{
 "cells": [
  {
   "cell_type": "code",
   "id": "initial_id",
   "metadata": {
    "collapsed": true,
    "ExecuteTime": {
     "end_time": "2024-10-30T23:33:44.996300Z",
     "start_time": "2024-10-30T23:33:44.812796Z"
    }
   },
   "source": [
    "import tensorflow as tf\n",
    "print('TensorFlow version: {}'.format(tf.__version__))"
   ],
   "outputs": [
    {
     "ename": "ImportError",
     "evalue": "Traceback (most recent call last):\n  File \"C:\\Users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages\\tensorflow\\python\\pywrap_tensorflow.py\", line 70, in <module>\n    from tensorflow.python._pywrap_tensorflow_internal import *\nImportError: DLL load failed while importing _pywrap_tensorflow_internal: The specified module could not be found.\n\n\nFailed to load the native TensorFlow runtime.\nSee https://www.tensorflow.org/install/errors for some common causes and solutions.\nIf you need help, create an issue at https://github.com/tensorflow/tensorflow/issues and include the entire stack trace above this error message.",
     "output_type": "error",
     "traceback": [
      "\u001B[1;31m---------------------------------------------------------------------------\u001B[0m",
      "\u001B[1;31mImportError\u001B[0m                               Traceback (most recent call last)",
      "File \u001B[1;32m~\\anaconda3\\envs\\medscript-ai\\lib\\site-packages\\tensorflow\\python\\pywrap_tensorflow.py:70\u001B[0m\n\u001B[0;32m     69\u001B[0m \u001B[38;5;28;01mtry\u001B[39;00m:\n\u001B[1;32m---> 70\u001B[0m   \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;21;01mtensorflow\u001B[39;00m\u001B[38;5;21;01m.\u001B[39;00m\u001B[38;5;21;01mpython\u001B[39;00m\u001B[38;5;21;01m.\u001B[39;00m\u001B[38;5;21;01m_pywrap_tensorflow_internal\u001B[39;00m \u001B[38;5;28;01mimport\u001B[39;00m \u001B[38;5;241m*\u001B[39m\n\u001B[0;32m     71\u001B[0m \u001B[38;5;66;03m# This try catch logic is because there is no bazel equivalent for py_extension.\u001B[39;00m\n\u001B[0;32m     72\u001B[0m \u001B[38;5;66;03m# Externally in opensource we must enable exceptions to load the shared object\u001B[39;00m\n\u001B[0;32m     73\u001B[0m \u001B[38;5;66;03m# by exposing the PyInit symbols with pybind. This error will only be\u001B[39;00m\n\u001B[0;32m     74\u001B[0m \u001B[38;5;66;03m# caught internally or if someone changes the name of the target _pywrap_tensorflow_internal.\u001B[39;00m\n\u001B[0;32m     75\u001B[0m \n\u001B[0;32m     76\u001B[0m \u001B[38;5;66;03m# This logic is used in other internal projects using py_extension.\u001B[39;00m\n",
      "\u001B[1;31mImportError\u001B[0m: DLL load failed while importing _pywrap_tensorflow_internal: The specified module could not be found.",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001B[1;31mImportError\u001B[0m                               Traceback (most recent call last)",
      "Cell \u001B[1;32mIn[3], line 1\u001B[0m\n\u001B[1;32m----> 1\u001B[0m \u001B[38;5;28;01mimport\u001B[39;00m \u001B[38;5;21;01mtensorflow\u001B[39;00m \u001B[38;5;28;01mas\u001B[39;00m \u001B[38;5;21;01mtf\u001B[39;00m\n\u001B[0;32m      2\u001B[0m \u001B[38;5;28mprint\u001B[39m(\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mTensorFlow version: \u001B[39m\u001B[38;5;132;01m{}\u001B[39;00m\u001B[38;5;124m'\u001B[39m\u001B[38;5;241m.\u001B[39mformat(tf\u001B[38;5;241m.\u001B[39m__version__))\n",
      "File \u001B[1;32m~\\anaconda3\\envs\\medscript-ai\\lib\\site-packages\\tensorflow\\__init__.py:40\u001B[0m\n\u001B[0;32m     37\u001B[0m _os\u001B[38;5;241m.\u001B[39menviron\u001B[38;5;241m.\u001B[39msetdefault(\u001B[38;5;124m\"\u001B[39m\u001B[38;5;124mENABLE_RUNTIME_UPTIME_TELEMETRY\u001B[39m\u001B[38;5;124m\"\u001B[39m, \u001B[38;5;124m\"\u001B[39m\u001B[38;5;124m1\u001B[39m\u001B[38;5;124m\"\u001B[39m)\n\u001B[0;32m     39\u001B[0m \u001B[38;5;66;03m# Do not remove this line; See https://github.com/tensorflow/tensorflow/issues/42596\u001B[39;00m\n\u001B[1;32m---> 40\u001B[0m \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;21;01mtensorflow\u001B[39;00m\u001B[38;5;21;01m.\u001B[39;00m\u001B[38;5;21;01mpython\u001B[39;00m \u001B[38;5;28;01mimport\u001B[39;00m pywrap_tensorflow \u001B[38;5;28;01mas\u001B[39;00m _pywrap_tensorflow  \u001B[38;5;66;03m# pylint: disable=unused-import\u001B[39;00m\n\u001B[0;32m     41\u001B[0m \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;21;01mtensorflow\u001B[39;00m\u001B[38;5;21;01m.\u001B[39;00m\u001B[38;5;21;01mpython\u001B[39;00m\u001B[38;5;21;01m.\u001B[39;00m\u001B[38;5;21;01mtools\u001B[39;00m \u001B[38;5;28;01mimport\u001B[39;00m module_util \u001B[38;5;28;01mas\u001B[39;00m _module_util\n\u001B[0;32m     42\u001B[0m \u001B[38;5;28;01mfrom\u001B[39;00m \u001B[38;5;21;01mtensorflow\u001B[39;00m\u001B[38;5;21;01m.\u001B[39;00m\u001B[38;5;21;01mpython\u001B[39;00m\u001B[38;5;21;01m.\u001B[39;00m\u001B[38;5;21;01mutil\u001B[39;00m\u001B[38;5;21;01m.\u001B[39;00m\u001B[38;5;21;01mlazy_loader\u001B[39;00m \u001B[38;5;28;01mimport\u001B[39;00m KerasLazyLoader \u001B[38;5;28;01mas\u001B[39;00m _KerasLazyLoader\n",
      "File \u001B[1;32m~\\anaconda3\\envs\\medscript-ai\\lib\\site-packages\\tensorflow\\python\\pywrap_tensorflow.py:85\u001B[0m\n\u001B[0;32m     83\u001B[0m     sys\u001B[38;5;241m.\u001B[39msetdlopenflags(_default_dlopen_flags)\n\u001B[0;32m     84\u001B[0m \u001B[38;5;28;01mexcept\u001B[39;00m \u001B[38;5;167;01mImportError\u001B[39;00m:\n\u001B[1;32m---> 85\u001B[0m   \u001B[38;5;28;01mraise\u001B[39;00m \u001B[38;5;167;01mImportError\u001B[39;00m(\n\u001B[0;32m     86\u001B[0m       \u001B[38;5;124mf\u001B[39m\u001B[38;5;124m'\u001B[39m\u001B[38;5;132;01m{\u001B[39;00mtraceback\u001B[38;5;241m.\u001B[39mformat_exc()\u001B[38;5;132;01m}\u001B[39;00m\u001B[38;5;124m'\u001B[39m\n\u001B[0;32m     87\u001B[0m       \u001B[38;5;124mf\u001B[39m\u001B[38;5;124m'\u001B[39m\u001B[38;5;130;01m\\n\u001B[39;00m\u001B[38;5;130;01m\\n\u001B[39;00m\u001B[38;5;124mFailed to load the native TensorFlow runtime.\u001B[39m\u001B[38;5;130;01m\\n\u001B[39;00m\u001B[38;5;124m'\u001B[39m\n\u001B[0;32m     88\u001B[0m       \u001B[38;5;124mf\u001B[39m\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mSee https://www.tensorflow.org/install/errors \u001B[39m\u001B[38;5;124m'\u001B[39m\n\u001B[0;32m     89\u001B[0m       \u001B[38;5;124mf\u001B[39m\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mfor some common causes and solutions.\u001B[39m\u001B[38;5;130;01m\\n\u001B[39;00m\u001B[38;5;124m'\u001B[39m\n\u001B[0;32m     90\u001B[0m       \u001B[38;5;124mf\u001B[39m\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mIf you need help, create an issue \u001B[39m\u001B[38;5;124m'\u001B[39m\n\u001B[0;32m     91\u001B[0m       \u001B[38;5;124mf\u001B[39m\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mat https://github.com/tensorflow/tensorflow/issues \u001B[39m\u001B[38;5;124m'\u001B[39m\n\u001B[0;32m     92\u001B[0m       \u001B[38;5;124mf\u001B[39m\u001B[38;5;124m'\u001B[39m\u001B[38;5;124mand include the entire stack trace above this error message.\u001B[39m\u001B[38;5;124m'\u001B[39m)\n",
      "\u001B[1;31mImportError\u001B[0m: Traceback (most recent call last):\n  File \"C:\\Users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages\\tensorflow\\python\\pywrap_tensorflow.py\", line 70, in <module>\n    from tensorflow.python._pywrap_tensorflow_internal import *\nImportError: DLL load failed while importing _pywrap_tensorflow_internal: The specified module could not be found.\n\n\nFailed to load the native TensorFlow runtime.\nSee https://www.tensorflow.org/install/errors for some common causes and solutions.\nIf you need help, create an issue at https://github.com/tensorflow/tensorflow/issues and include the entire stack trace above this error message."
     ]
    }
   ],
   "execution_count": 3
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-30T19:39:03.499876Z",
     "start_time": "2024-10-30T19:32:55.912821Z"
    }
   },
   "cell_type": "code",
   "source": "!pip install -U tfx",
   "id": "f3a7bf0592f822cf",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting tfx\n",
      "  Using cached tfx-1.15.1-py3-none-any.whl.metadata (23 kB)\n",
      "Collecting ml-pipelines-sdk==1.15.1 (from tfx)\n",
      "  Using cached ml_pipelines_sdk-1.15.1-py3-none-any.whl.metadata (2.3 kB)\n",
      "Collecting absl-py<2.0.0,>=0.9 (from tfx)\n",
      "  Using cached absl_py-1.4.0-py3-none-any.whl.metadata (2.3 kB)\n",
      "INFO: pip is looking at multiple versions of tfx to determine which version is compatible with other requirements. This could take a while.\n",
      "Collecting tfx\n",
      "  Using cached tfx-1.15.0-py3-none-any.whl.metadata (23 kB)\n",
      "Collecting ml-pipelines-sdk==1.15.0 (from tfx)\n",
      "  Using cached ml_pipelines_sdk-1.15.0-py3-none-any.whl.metadata (2.3 kB)\n",
      "Collecting tfx\n",
      "  Using cached tfx-1.14.0-py3-none-any.whl.metadata (22 kB)\n",
      "Collecting ml-pipelines-sdk==1.14.0 (from tfx)\n",
      "  Using cached ml_pipelines_sdk-1.14.0-py3-none-any.whl.metadata (2.3 kB)\n",
      "Collecting ml-metadata<1.15.0,>=1.14.0 (from tfx)\n",
      "  Using cached ml_metadata-1.14.0-cp310-cp310-win_amd64.whl.metadata (5.1 kB)\n",
      "Collecting packaging<21,>=20 (from tfx)\n",
      "  Using cached packaging-20.9-py2.py3-none-any.whl.metadata (13 kB)\n",
      "Collecting portpicker<2,>=1.3.1 (from tfx)\n",
      "  Using cached portpicker-1.6.0-py3-none-any.whl.metadata (1.5 kB)\n",
      "Requirement already satisfied: protobuf<5,>=3.20.3 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tfx) (3.20.3)\n",
      "Collecting docker<5,>=4.1 (from tfx)\n",
      "  Using cached docker-4.4.4-py2.py3-none-any.whl.metadata (3.8 kB)\n",
      "Collecting google-apitools<1,>=0.5 (from tfx)\n",
      "  Using cached google_apitools-0.5.32-py3-none-any.whl.metadata (2.3 kB)\n",
      "Collecting google-api-python-client<2,>=1.8 (from tfx)\n",
      "  Using cached google_api_python_client-1.12.11-py2.py3-none-any.whl.metadata (4.2 kB)\n",
      "Requirement already satisfied: jinja2<4,>=2.7.3 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tfx) (3.1.4)\n",
      "Requirement already satisfied: typing-extensions<5,>=3.10.0.2 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tfx) (4.11.0)\n",
      "Collecting apache-beam<3,>=2.47 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached apache_beam-2.60.0-cp310-cp310-win_amd64.whl.metadata (7.7 kB)\n",
      "Collecting attrs<22,>=19.3.0 (from tfx)\n",
      "  Using cached attrs-21.4.0-py2.py3-none-any.whl.metadata (9.8 kB)\n",
      "Requirement already satisfied: click<9,>=7 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tfx) (8.1.7)\n",
      "Collecting google-api-core<3 (from tfx)\n",
      "  Using cached google_api_core-2.22.0-py3-none-any.whl.metadata (2.9 kB)\n",
      "Collecting google-cloud-aiplatform<2,>=1.6.2 (from tfx)\n",
      "  Using cached google_cloud_aiplatform-1.71.0-py2.py3-none-any.whl.metadata (32 kB)\n",
      "Collecting google-cloud-bigquery<3,>=2.26.0 (from tfx)\n",
      "  Using cached google_cloud_bigquery-2.34.4-py2.py3-none-any.whl.metadata (7.9 kB)\n",
      "Requirement already satisfied: grpcio<2,>=1.28.1 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tfx) (1.67.1)\n",
      "Collecting keras-tuner<2,>=1.0.4 (from tfx)\n",
      "  Using cached keras_tuner-1.4.7-py3-none-any.whl.metadata (5.4 kB)\n",
      "Collecting kubernetes<13,>=10.0.1 (from tfx)\n",
      "  Using cached kubernetes-12.0.1-py2.py3-none-any.whl.metadata (1.5 kB)\n",
      "Requirement already satisfied: numpy<2,>=1.16 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tfx) (1.26.4)\n",
      "Collecting pyarrow<11,>=10 (from tfx)\n",
      "  Using cached pyarrow-10.0.1-cp310-cp310-win_amd64.whl.metadata (3.1 kB)\n",
      "Requirement already satisfied: pyyaml<7,>=6 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tfx) (6.0.2)\n",
      "Collecting tensorflow<2.14,>=2.13.0 (from tfx)\n",
      "  Using cached tensorflow-2.13.1-cp310-cp310-win_amd64.whl.metadata (2.6 kB)\n",
      "Collecting tensorflow-hub<0.14,>=0.9.0 (from tfx)\n",
      "  Using cached tensorflow_hub-0.13.0-py2.py3-none-any.whl.metadata (1.7 kB)\n",
      "Collecting tensorflow-data-validation<1.15.0,>=1.14.0 (from tfx)\n",
      "  Using cached tensorflow_data_validation-1.14.0-cp310-cp310-win_amd64.whl.metadata (18 kB)\n",
      "Collecting tensorflow-model-analysis<0.46.0,>=0.45.0 (from tfx)\n",
      "  Using cached tensorflow_model_analysis-0.45.0-py3-none-any.whl.metadata (19 kB)\n",
      "Requirement already satisfied: tensorflow-serving-api!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,<3,>=1.15 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tfx) (2.17.0)\n",
      "Collecting tensorflow-transform<1.15.0,>=1.14.0 (from tfx)\n",
      "  Using cached tensorflow_transform-1.14.0-py3-none-any.whl.metadata (12 kB)\n",
      "Collecting tfx-bsl<1.15.0,>=1.14.0 (from tfx)\n",
      "  Using cached tfx_bsl-1.14.0-cp310-cp310-win_amd64.whl.metadata (10 kB)\n",
      "Collecting crcmod<2.0,>=1.7 (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached crcmod-1.7.tar.gz (89 kB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Collecting orjson<4,>=3.9.7 (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached orjson-3.10.10-cp310-none-win_amd64.whl.metadata (51 kB)\n",
      "Collecting dill<0.3.2,>=0.3.1.1 (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached dill-0.3.1.1.tar.gz (151 kB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Collecting cloudpickle~=2.2.1 (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached cloudpickle-2.2.1-py3-none-any.whl.metadata (6.9 kB)\n",
      "Collecting fastavro<2,>=0.23.6 (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached fastavro-1.9.7-cp310-cp310-win_amd64.whl.metadata (5.6 kB)\n",
      "Collecting fasteners<1.0,>=0.3 (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached fasteners-0.19-py3-none-any.whl.metadata (4.9 kB)\n",
      "Collecting grpcio<2,>=1.28.1 (from tfx)\n",
      "  Using cached grpcio-1.65.5-cp310-cp310-win_amd64.whl.metadata (3.4 kB)\n",
      "Collecting hdfs<3.0.0,>=2.1.0 (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached hdfs-2.7.3.tar.gz (43 kB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Collecting httplib2<0.23.0,>=0.8 (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached httplib2-0.22.0-py3-none-any.whl.metadata (2.6 kB)\n",
      "Requirement already satisfied: jsonschema<5.0.0,>=4.0.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx) (4.23.0)\n",
      "Collecting jsonpickle<4.0.0,>=3.0.0 (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached jsonpickle-3.3.0-py3-none-any.whl.metadata (8.3 kB)\n",
      "Collecting objsize<0.8.0,>=0.6.1 (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached objsize-0.7.0-py3-none-any.whl.metadata (12 kB)\n",
      "INFO: pip is looking at multiple versions of apache-beam to determine which version is compatible with other requirements. This could take a while.\n",
      "Collecting apache-beam<3,>=2.47 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached apache_beam-2.59.0-cp310-cp310-win_amd64.whl.metadata (7.6 kB)\n",
      "  Using cached apache_beam-2.58.1-cp310-cp310-win_amd64.whl.metadata (7.6 kB)\n",
      "  Using cached apache_beam-2.58.0-cp310-cp310-win_amd64.whl.metadata (7.8 kB)\n",
      "  Using cached apache_beam-2.57.0-cp310-cp310-win_amd64.whl.metadata (7.8 kB)\n",
      "  Using cached apache_beam-2.56.0-cp310-cp310-win_amd64.whl.metadata (6.9 kB)\n",
      "  Using cached apache_beam-2.55.1-cp310-cp310-win_amd64.whl.metadata (6.8 kB)\n",
      "Collecting js2py<1,>=0.74 (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached Js2Py-0.74-py3-none-any.whl.metadata (868 bytes)\n",
      "Collecting apache-beam<3,>=2.47 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached apache_beam-2.55.0-cp310-cp310-win_amd64.whl.metadata (6.8 kB)\n",
      "INFO: pip is still looking at multiple versions of apache-beam to determine which version is compatible with other requirements. This could take a while.\n",
      "  Using cached apache_beam-2.54.0-cp310-cp310-win_amd64.whl.metadata (6.8 kB)\n",
      "Collecting numpy<2,>=1.16 (from tfx)\n",
      "  Using cached numpy-1.24.4-cp310-cp310-win_amd64.whl.metadata (5.6 kB)\n",
      "Collecting apache-beam<3,>=2.47 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached apache_beam-2.53.0-cp310-cp310-win_amd64.whl.metadata (6.7 kB)\n",
      "Collecting objsize<0.7.0,>=0.6.1 (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached objsize-0.6.1-py3-none-any.whl.metadata (10 kB)\n",
      "Collecting apache-beam<3,>=2.47 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached apache_beam-2.52.0-cp310-cp310-win_amd64.whl.metadata (6.4 kB)\n",
      "  Using cached apache_beam-2.51.0-cp310-cp310-win_amd64.whl.metadata (6.1 kB)\n",
      "  Using cached apache_beam-2.50.0-cp310-cp310-win_amd64.whl.metadata (6.0 kB)\n",
      "Collecting pymongo<5.0.0,>=3.8.0 (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached pymongo-4.10.1-cp310-cp310-win_amd64.whl.metadata (22 kB)\n",
      "Collecting proto-plus<2,>=1.7.1 (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached proto_plus-1.25.0-py3-none-any.whl.metadata (2.2 kB)\n",
      "Collecting pydot<2,>=1.2.0 (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached pydot-1.4.2-py2.py3-none-any.whl.metadata (8.0 kB)\n",
      "Requirement already satisfied: python-dateutil<3,>=2.8.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx) (2.9.0.post0)\n",
      "Requirement already satisfied: pytz>=2018.3 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx) (2024.1)\n",
      "Collecting regex>=2020.6.8 (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached regex-2024.9.11-cp310-cp310-win_amd64.whl.metadata (41 kB)\n",
      "Requirement already satisfied: requests<3.0.0,>=2.24.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx) (2.32.3)\n",
      "Collecting zstandard<1,>=0.18.0 (from apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached zstandard-0.23.0-cp310-cp310-win_amd64.whl.metadata (3.0 kB)\n",
      "INFO: pip is looking at multiple versions of apache-beam[gcp] to determine which version is compatible with other requirements. This could take a while.\n",
      "INFO: pip is still looking at multiple versions of apache-beam[gcp] to determine which version is compatible with other requirements. This could take a while.\n",
      "Collecting cachetools<6,>=3.1.0 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached cachetools-5.5.0-py3-none-any.whl.metadata (5.3 kB)\n",
      "Collecting google-apitools<1,>=0.5 (from tfx)\n",
      "  Using cached google-apitools-0.5.31.tar.gz (173 kB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Collecting google-auth<3,>=1.18.0 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached google_auth-2.35.0-py2.py3-none-any.whl.metadata (4.7 kB)\n",
      "Collecting google-auth-httplib2<0.2.0,>=0.1.0 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached google_auth_httplib2-0.1.1-py2.py3-none-any.whl.metadata (2.1 kB)\n",
      "Collecting google-cloud-datastore<3,>=2.0.0 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached google_cloud_datastore-2.20.1-py2.py3-none-any.whl.metadata (5.6 kB)\n",
      "Collecting google-cloud-pubsub<3,>=2.1.0 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached google_cloud_pubsub-2.26.1-py2.py3-none-any.whl.metadata (9.6 kB)\n",
      "Collecting google-cloud-pubsublite<2,>=1.2.0 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached google_cloud_pubsublite-1.11.1-py2.py3-none-any.whl.metadata (5.6 kB)\n",
      "Collecting google-cloud-bigquery-storage<3,>=2.6.3 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached google_cloud_bigquery_storage-2.27.0-py2.py3-none-any.whl.metadata (5.6 kB)\n",
      "Collecting google-cloud-core<3,>=2.0.0 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached google_cloud_core-2.4.1-py2.py3-none-any.whl.metadata (2.7 kB)\n",
      "Collecting google-cloud-bigtable<3,>=2.19.0 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached google_cloud_bigtable-2.26.0-py2.py3-none-any.whl.metadata (5.4 kB)\n",
      "Collecting google-cloud-spanner<4,>=3.0.0 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached google_cloud_spanner-3.49.1-py2.py3-none-any.whl.metadata (10 kB)\n",
      "Collecting google-cloud-dlp<4,>=3.0.0 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached google_cloud_dlp-3.25.0-py2.py3-none-any.whl.metadata (5.4 kB)\n",
      "Collecting google-cloud-language<3,>=2.0 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached google_cloud_language-2.15.0-py2.py3-none-any.whl.metadata (5.5 kB)\n",
      "Collecting google-cloud-videointelligence<3,>=2.0 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached google_cloud_videointelligence-2.14.0-py2.py3-none-any.whl.metadata (5.8 kB)\n",
      "Collecting google-cloud-vision<4,>=2 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached google_cloud_vision-3.8.0-py2.py3-none-any.whl.metadata (5.3 kB)\n",
      "Collecting google-cloud-recommendations-ai<0.11.0,>=0.1.0 (from apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached google_cloud_recommendations_ai-0.10.13-py2.py3-none-any.whl.metadata (5.4 kB)\n",
      "Requirement already satisfied: colorama in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from click<9,>=7->tfx) (0.4.6)\n",
      "Requirement already satisfied: six>=1.4.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from docker<5,>=4.1->tfx) (1.16.0)\n",
      "Requirement already satisfied: websocket-client>=0.32.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from docker<5,>=4.1->tfx) (1.8.0)\n",
      "INFO: pip is looking at multiple versions of docker to determine which version is compatible with other requirements. This could take a while.\n",
      "Collecting docker<5,>=4.1 (from tfx)\n",
      "  Using cached docker-4.4.3-py2.py3-none-any.whl.metadata (3.8 kB)\n",
      "  Using cached docker-4.4.2-py2.py3-none-any.whl.metadata (3.8 kB)\n",
      "  Using cached docker-4.4.1-py2.py3-none-any.whl.metadata (3.7 kB)\n",
      "  Using cached docker-4.4.0-py2.py3-none-any.whl.metadata (3.7 kB)\n",
      "  Using cached docker-4.3.1-py2.py3-none-any.whl.metadata (3.7 kB)\n",
      "  Using cached docker-4.3.0-py2.py3-none-any.whl.metadata (3.7 kB)\n",
      "  Using cached docker-4.2.2-py2.py3-none-any.whl.metadata (3.8 kB)\n",
      "Collecting pypiwin32==223 (from docker<5,>=4.1->tfx)\n",
      "  Using cached pypiwin32-223-py3-none-any.whl.metadata (236 bytes)\n",
      "Requirement already satisfied: pywin32>=223 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from pypiwin32==223->docker<5,>=4.1->tfx) (305.1)\n",
      "Collecting googleapis-common-protos<2.0.dev0,>=1.56.2 (from google-api-core<3->tfx)\n",
      "  Using cached googleapis_common_protos-1.65.0-py2.py3-none-any.whl.metadata (1.5 kB)\n",
      "Collecting uritemplate<4dev,>=3.0.0 (from google-api-python-client<2,>=1.8->tfx)\n",
      "  Using cached uritemplate-3.0.1-py2.py3-none-any.whl.metadata (4.6 kB)\n",
      "Collecting oauth2client>=1.4.12 (from google-apitools<1,>=0.5->tfx)\n",
      "  Using cached oauth2client-4.1.3-py2.py3-none-any.whl.metadata (1.2 kB)\n",
      "Collecting google-cloud-storage<3.0.0dev,>=1.32.0 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached google_cloud_storage-2.18.2-py2.py3-none-any.whl.metadata (9.1 kB)\n",
      "Collecting google-cloud-resource-manager<3.0.0dev,>=1.3.3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached google_cloud_resource_manager-1.13.0-py2.py3-none-any.whl.metadata (5.4 kB)\n",
      "Collecting shapely<3.0.0dev (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached shapely-2.0.6-cp310-cp310-win_amd64.whl.metadata (7.2 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.9.2-py3-none-any.whl.metadata (149 kB)\n",
      "Requirement already satisfied: docstring-parser<1 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from google-cloud-aiplatform<2,>=1.6.2->tfx) (0.16)\n",
      "Collecting google-resumable-media<3.0dev,>=0.6.0 (from google-cloud-bigquery<3,>=2.26.0->tfx)\n",
      "  Using cached google_resumable_media-2.7.2-py2.py3-none-any.whl.metadata (2.2 kB)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jinja2<4,>=2.7.3->tfx) (2.1.3)\n",
      "Requirement already satisfied: keras in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from keras-tuner<2,>=1.0.4->tfx) (3.6.0)\n",
      "Collecting kt-legacy (from keras-tuner<2,>=1.0.4->tfx)\n",
      "  Using cached kt_legacy-1.0.5-py3-none-any.whl.metadata (221 bytes)\n",
      "Requirement already satisfied: certifi>=14.05.14 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from kubernetes<13,>=10.0.1->tfx) (2024.8.30)\n",
      "Requirement already satisfied: setuptools>=21.0.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from kubernetes<13,>=10.0.1->tfx) (75.1.0)\n",
      "Collecting requests-oauthlib (from kubernetes<13,>=10.0.1->tfx)\n",
      "  Using cached requests_oauthlib-2.0.0-py2.py3-none-any.whl.metadata (11 kB)\n",
      "Requirement already satisfied: urllib3>=1.24.2 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from kubernetes<13,>=10.0.1->tfx) (2.2.3)\n",
      "Requirement already satisfied: pyparsing>=2.0.2 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from packaging<21,>=20->tfx) (3.1.2)\n",
      "Requirement already satisfied: psutil in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from portpicker<2,>=1.3.1->tfx) (5.9.0)\n",
      "Collecting tensorflow-intel==2.13.1 (from tensorflow<2.14,>=2.13.0->tfx)\n",
      "  Using cached tensorflow_intel-2.13.1-cp310-cp310-win_amd64.whl.metadata (4.1 kB)\n",
      "Requirement already satisfied: astunparse>=1.6.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tensorflow-intel==2.13.1->tensorflow<2.14,>=2.13.0->tfx) (1.6.3)\n",
      "Requirement already satisfied: flatbuffers>=23.1.21 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tensorflow-intel==2.13.1->tensorflow<2.14,>=2.13.0->tfx) (24.3.25)\n",
      "Collecting gast<=0.4.0,>=0.2.1 (from tensorflow-intel==2.13.1->tensorflow<2.14,>=2.13.0->tfx)\n",
      "  Using cached gast-0.4.0-py3-none-any.whl.metadata (1.1 kB)\n",
      "Requirement already satisfied: google-pasta>=0.1.1 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tensorflow-intel==2.13.1->tensorflow<2.14,>=2.13.0->tfx) (0.2.0)\n",
      "Requirement already satisfied: h5py>=2.9.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tensorflow-intel==2.13.1->tensorflow<2.14,>=2.13.0->tfx) (3.12.1)\n",
      "Requirement already satisfied: libclang>=13.0.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tensorflow-intel==2.13.1->tensorflow<2.14,>=2.13.0->tfx) (18.1.1)\n",
      "Collecting numpy<2,>=1.16 (from tfx)\n",
      "  Using cached numpy-1.24.3-cp310-cp310-win_amd64.whl.metadata (5.6 kB)\n",
      "Requirement already satisfied: opt-einsum>=2.3.2 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tensorflow-intel==2.13.1->tensorflow<2.14,>=2.13.0->tfx) (3.4.0)\n",
      "Requirement already satisfied: termcolor>=1.1.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tensorflow-intel==2.13.1->tensorflow<2.14,>=2.13.0->tfx) (2.5.0)\n",
      "Collecting typing-extensions<5,>=3.10.0.2 (from tfx)\n",
      "  Using cached typing_extensions-4.5.0-py3-none-any.whl.metadata (8.5 kB)\n",
      "Requirement already satisfied: wrapt>=1.11.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tensorflow-intel==2.13.1->tensorflow<2.14,>=2.13.0->tfx) (1.16.0)\n",
      "Collecting tensorboard<2.14,>=2.13 (from tensorflow-intel==2.13.1->tensorflow<2.14,>=2.13.0->tfx)\n",
      "  Using cached tensorboard-2.13.0-py3-none-any.whl.metadata (1.8 kB)\n",
      "Collecting tensorflow-estimator<2.14,>=2.13.0 (from tensorflow-intel==2.13.1->tensorflow<2.14,>=2.13.0->tfx)\n",
      "  Using cached tensorflow_estimator-2.13.0-py2.py3-none-any.whl.metadata (1.3 kB)\n",
      "Collecting keras (from keras-tuner<2,>=1.0.4->tfx)\n",
      "  Using cached keras-2.13.1-py3-none-any.whl.metadata (2.4 kB)\n",
      "Requirement already satisfied: tensorflow-io-gcs-filesystem>=0.23.1 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tensorflow-intel==2.13.1->tensorflow<2.14,>=2.13.0->tfx) (0.31.0)\n",
      "Requirement already satisfied: joblib>=1.2.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tensorflow-data-validation<1.15.0,>=1.14.0->tfx) (1.4.2)\n",
      "Collecting pandas<2,>=1.0 (from tensorflow-data-validation<1.15.0,>=1.14.0->tfx)\n",
      "  Using cached pandas-1.5.3-cp310-cp310-win_amd64.whl.metadata (12 kB)\n",
      "Collecting pyfarmhash<0.4,>=0.2.2 (from tensorflow-data-validation<1.15.0,>=1.14.0->tfx)\n",
      "  Using cached pyfarmhash-0.3.2-cp310-cp310-win_amd64.whl.metadata (1.8 kB)\n",
      "Collecting tensorflow-metadata<1.15,>=1.14.0 (from tensorflow-data-validation<1.15.0,>=1.14.0->tfx)\n",
      "  Using cached tensorflow_metadata-1.14.0-py3-none-any.whl.metadata (2.1 kB)\n",
      "Collecting ipython<8,>=7 (from tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached ipython-7.34.0-py3-none-any.whl.metadata (4.3 kB)\n",
      "Collecting ipywidgets<8,>=7 (from tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached ipywidgets-7.8.5-py2.py3-none-any.whl.metadata (1.9 kB)\n",
      "Requirement already satisfied: pillow>=9.4.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (10.4.0)\n",
      "Requirement already satisfied: scipy<2,>=1.4.1 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (1.13.1)\n",
      "INFO: pip is looking at multiple versions of tensorflow-serving-api to determine which version is compatible with other requirements. This could take a while.\n",
      "Collecting tensorflow-serving-api!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,!=2.8.*,<3,>=1.15 (from tfx)\n",
      "  Downloading tensorflow_serving_api-2.17.1-py2.py3-none-any.whl.metadata (1.8 kB)\n",
      "  Using cached tensorflow_serving_api-2.16.1-py2.py3-none-any.whl.metadata (1.8 kB)\n",
      "  Using cached tensorflow_serving_api-2.15.1-py2.py3-none-any.whl.metadata (1.8 kB)\n",
      "  Using cached tensorflow_serving_api-2.14.1-py2.py3-none-any.whl.metadata (1.8 kB)\n",
      "  Using cached tensorflow_serving_api-2.14.0-py2.py3-none-any.whl.metadata (1.8 kB)\n",
      "  Using cached tensorflow_serving_api-2.13.1-py2.py3-none-any.whl.metadata (1.8 kB)\n",
      "Collecting grpcio-status<2.0.dev0,>=1.33.2 (from google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,<3.0.0dev,>=1.34.1->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached grpcio_status-1.67.1-py3-none-any.whl.metadata (1.1 kB)\n",
      "Collecting pyasn1-modules>=0.2.1 (from google-auth<3,>=1.18.0->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached pyasn1_modules-0.4.1-py3-none-any.whl.metadata (3.5 kB)\n",
      "Collecting rsa<5,>=3.1.4 (from google-auth<3,>=1.18.0->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached rsa-4.9-py3-none-any.whl.metadata (4.2 kB)\n",
      "Collecting grpc-google-iam-v1<1.0.0dev,>=0.12.4 (from google-cloud-bigtable<3,>=2.19.0->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached grpc_google_iam_v1-0.13.1-py2.py3-none-any.whl.metadata (3.3 kB)\n",
      "Collecting opentelemetry-api>=1.27.0 (from google-cloud-pubsub<3,>=2.1.0->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached opentelemetry_api-1.27.0-py3-none-any.whl.metadata (1.4 kB)\n",
      "Collecting opentelemetry-sdk>=1.27.0 (from google-cloud-pubsub<3,>=2.1.0->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached opentelemetry_sdk-1.27.0-py3-none-any.whl.metadata (1.5 kB)\n",
      "Requirement already satisfied: overrides<8.0.0,>=6.0.1 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from google-cloud-pubsublite<2,>=1.2.0->apache-beam[gcp]<3,>=2.47->tfx) (7.4.0)\n",
      "Collecting sqlparse>=0.4.4 (from google-cloud-spanner<4,>=3.0.0->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached sqlparse-0.5.1-py3-none-any.whl.metadata (3.9 kB)\n",
      "Collecting grpc-interceptor>=0.15.4 (from google-cloud-spanner<4,>=3.0.0->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached grpc_interceptor-0.15.4-py3-none-any.whl.metadata (8.4 kB)\n",
      "Collecting google-crc32c<2.0dev,>=1.0 (from google-cloud-storage<3.0.0dev,>=1.32.0->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached google_crc32c-1.6.0-cp310-cp310-win_amd64.whl.metadata (2.4 kB)\n",
      "Collecting docopt (from hdfs<3.0.0,>=2.1.0->apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached docopt-0.6.2.tar.gz (25 kB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Requirement already satisfied: jedi>=0.16 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from ipython<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.19.1)\n",
      "Requirement already satisfied: decorator in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from ipython<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (5.1.1)\n",
      "Collecting pickleshare (from ipython<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached pickleshare-0.7.5-py2.py3-none-any.whl.metadata (1.5 kB)\n",
      "Requirement already satisfied: traitlets>=4.2 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from ipython<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (5.14.3)\n",
      "Requirement already satisfied: prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from ipython<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (3.0.43)\n",
      "Requirement already satisfied: pygments in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from ipython<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (2.15.1)\n",
      "Collecting backcall (from ipython<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached backcall-0.2.0-py2.py3-none-any.whl.metadata (2.0 kB)\n",
      "Requirement already satisfied: matplotlib-inline in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from ipython<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.1.6)\n",
      "Requirement already satisfied: comm>=0.1.3 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.2.1)\n",
      "Collecting ipython-genutils~=0.2.0 (from ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached ipython_genutils-0.2.0-py2.py3-none-any.whl.metadata (755 bytes)\n",
      "Collecting widgetsnbextension~=3.6.10 (from ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached widgetsnbextension-3.6.10-py2.py3-none-any.whl.metadata (1.3 kB)\n",
      "Collecting jupyterlab-widgets<3,>=1.0.0 (from ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached jupyterlab_widgets-1.1.11-py3-none-any.whl.metadata (3.7 kB)\n",
      "Collecting pyasn1>=0.1.7 (from oauth2client>=1.4.12->google-apitools<1,>=0.5->tfx)\n",
      "  Using cached pyasn1-0.6.1-py3-none-any.whl.metadata (8.4 kB)\n",
      "Collecting annotated-types>=0.6.0 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached annotated_types-0.7.0-py3-none-any.whl.metadata (15 kB)\n",
      "Collecting pydantic-core==2.23.4 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.23.4-cp310-none-win_amd64.whl.metadata (6.7 kB)\n",
      "INFO: pip is looking at multiple versions of pydantic to determine which version is compatible with other requirements. This could take a while.\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.9.1-py3-none-any.whl.metadata (146 kB)\n",
      "Collecting pydantic-core==2.23.3 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.23.3-cp310-none-win_amd64.whl.metadata (6.7 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.9.0-py3-none-any.whl.metadata (146 kB)\n",
      "Collecting pydantic-core==2.23.2 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.23.2-cp310-none-win_amd64.whl.metadata (6.7 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.8.2-py3-none-any.whl.metadata (125 kB)\n",
      "Collecting pydantic-core==2.20.1 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.20.1-cp310-none-win_amd64.whl.metadata (6.7 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.8.1-py3-none-any.whl.metadata (124 kB)\n",
      "  Using cached pydantic-2.8.0-py3-none-any.whl.metadata (123 kB)\n",
      "Collecting pydantic-core==2.20.0 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.20.0-cp310-none-win_amd64.whl.metadata (6.7 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.7.4-py3-none-any.whl.metadata (109 kB)\n",
      "Collecting pydantic-core==2.18.4 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.18.4-cp310-none-win_amd64.whl.metadata (6.7 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.7.3-py3-none-any.whl.metadata (108 kB)\n",
      "INFO: pip is still looking at multiple versions of pydantic to determine which version is compatible with other requirements. This could take a while.\n",
      "  Using cached pydantic-2.7.2-py3-none-any.whl.metadata (108 kB)\n",
      "Collecting pydantic-core==2.18.3 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.18.3-cp310-none-win_amd64.whl.metadata (6.7 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.7.1-py3-none-any.whl.metadata (107 kB)\n",
      "Collecting pydantic-core==2.18.2 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.18.2-cp310-none-win_amd64.whl.metadata (6.7 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.7.0-py3-none-any.whl.metadata (103 kB)\n",
      "Collecting pydantic-core==2.18.1 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.18.1-cp310-none-win_amd64.whl.metadata (6.7 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.6.4-py3-none-any.whl.metadata (85 kB)\n",
      "Collecting pydantic-core==2.16.3 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.16.3-cp310-none-win_amd64.whl.metadata (6.6 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.6.3-py3-none-any.whl.metadata (84 kB)\n",
      "INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. See https://pip.pypa.io/warnings/backtracking for guidance. If you want to abort this run, press Ctrl + C.\n",
      "  Using cached pydantic-2.6.2-py3-none-any.whl.metadata (83 kB)\n",
      "  Using cached pydantic-2.6.1-py3-none-any.whl.metadata (83 kB)\n",
      "Collecting pydantic-core==2.16.2 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.16.2-cp310-none-win_amd64.whl.metadata (6.6 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.6.0-py3-none-any.whl.metadata (81 kB)\n",
      "Collecting pydantic-core==2.16.1 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.16.1-cp310-none-win_amd64.whl.metadata (6.6 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.5.3-py3-none-any.whl.metadata (65 kB)\n",
      "Collecting pydantic-core==2.14.6 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.14.6-cp310-none-win_amd64.whl.metadata (6.6 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.5.2-py3-none-any.whl.metadata (65 kB)\n",
      "Collecting pydantic-core==2.14.5 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.14.5-cp310-none-win_amd64.whl.metadata (6.6 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.5.1-py3-none-any.whl.metadata (64 kB)\n",
      "Collecting pydantic-core==2.14.3 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.14.3-cp310-none-win_amd64.whl.metadata (6.6 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.5.0-py3-none-any.whl.metadata (174 kB)\n",
      "Collecting pydantic-core==2.14.1 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.14.1-cp310-none-win_amd64.whl.metadata (6.6 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.4.2-py3-none-any.whl.metadata (158 kB)\n",
      "Collecting pydantic-core==2.10.1 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.10.1-cp310-none-win_amd64.whl.metadata (6.6 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.4.1-py3-none-any.whl.metadata (157 kB)\n",
      "  Using cached pydantic-2.4.0-py3-none-any.whl.metadata (156 kB)\n",
      "Collecting pydantic-core==2.10.0 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.10.0-cp310-none-win_amd64.whl.metadata (6.6 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.3.0-py3-none-any.whl.metadata (148 kB)\n",
      "Collecting pydantic-core==2.6.3 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.6.3-cp310-none-win_amd64.whl.metadata (6.6 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.2.1-py3-none-any.whl.metadata (145 kB)\n",
      "Collecting pydantic-core==2.6.1 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.6.1-cp310-none-win_amd64.whl.metadata (6.6 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.2.0-py3-none-any.whl.metadata (145 kB)\n",
      "Collecting pydantic-core==2.6.0 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.6.0-cp310-none-win_amd64.whl.metadata (6.6 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.1.1-py3-none-any.whl.metadata (136 kB)\n",
      "Collecting pydantic-core==2.4.0 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.4.0-cp310-none-win_amd64.whl.metadata (6.6 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.1.0-py3-none-any.whl.metadata (136 kB)\n",
      "  Using cached pydantic-2.0.3-py3-none-any.whl.metadata (128 kB)\n",
      "Collecting pydantic-core==2.3.0 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.3.0-cp310-none-win_amd64.whl.metadata (5.2 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.0.2-py3-none-any.whl.metadata (119 kB)\n",
      "Collecting pydantic-core==2.1.2 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.1.2-cp310-none-win_amd64.whl.metadata (5.2 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.0.1-py3-none-any.whl.metadata (119 kB)\n",
      "Collecting pydantic-core==2.0.2 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.0.2-cp310-none-win_amd64.whl.metadata (5.2 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-2.0-py3-none-any.whl.metadata (117 kB)\n",
      "Collecting pydantic-core==2.0.1 (from pydantic<3->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic_core-2.0.1-cp310-none-win_amd64.whl.metadata (5.3 kB)\n",
      "Collecting pydantic<3 (from google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached pydantic-1.10.18-cp310-cp310-win_amd64.whl.metadata (153 kB)\n",
      "Collecting dnspython<3.0.0,>=1.16.0 (from pymongo<5.0.0,>=3.8.0->apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached dnspython-2.7.0-py3-none-any.whl.metadata (5.8 kB)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from requests<3.0.0,>=2.24.0->apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx) (3.3.2)\n",
      "Requirement already satisfied: idna<4,>=2.5 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from requests<3.0.0,>=2.24.0->apache-beam<3,>=2.47->apache-beam[gcp]<3,>=2.47->tfx) (3.10)\n",
      "Collecting oauthlib>=3.0.0 (from requests-oauthlib->kubernetes<13,>=10.0.1->tfx)\n",
      "  Using cached oauthlib-3.2.2-py3-none-any.whl.metadata (7.5 kB)\n",
      "Requirement already satisfied: wheel<1.0,>=0.23.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from astunparse>=1.6.0->tensorflow-intel==2.13.1->tensorflow<2.14,>=2.13.0->tfx) (0.44.0)\n",
      "INFO: pip is looking at multiple versions of grpcio-status to determine which version is compatible with other requirements. This could take a while.\n",
      "Collecting grpcio-status<2.0.dev0,>=1.33.2 (from google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.*,!=2.4.*,!=2.5.*,!=2.6.*,!=2.7.*,<3.0.0dev,>=1.34.1->google-cloud-aiplatform<2,>=1.6.2->tfx)\n",
      "  Using cached grpcio_status-1.67.0-py3-none-any.whl.metadata (1.1 kB)\n",
      "  Using cached grpcio_status-1.66.2-py3-none-any.whl.metadata (1.1 kB)\n",
      "  Using cached grpcio_status-1.66.1-py3-none-any.whl.metadata (1.1 kB)\n",
      "  Using cached grpcio_status-1.66.0-py3-none-any.whl.metadata (1.1 kB)\n",
      "  Using cached grpcio_status-1.65.5-py3-none-any.whl.metadata (1.1 kB)\n",
      "  Using cached grpcio_status-1.65.4-py3-none-any.whl.metadata (1.1 kB)\n",
      "  Using cached grpcio_status-1.65.2-py3-none-any.whl.metadata (1.1 kB)\n",
      "INFO: pip is still looking at multiple versions of grpcio-status to determine which version is compatible with other requirements. This could take a while.\n",
      "  Using cached grpcio_status-1.65.1-py3-none-any.whl.metadata (1.1 kB)\n",
      "  Using cached grpcio_status-1.64.3-py3-none-any.whl.metadata (1.1 kB)\n",
      "  Using cached grpcio_status-1.64.1-py3-none-any.whl.metadata (1.1 kB)\n",
      "  Using cached grpcio_status-1.64.0-py3-none-any.whl.metadata (1.1 kB)\n",
      "  Using cached grpcio_status-1.63.2-py3-none-any.whl.metadata (1.1 kB)\n",
      "INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. See https://pip.pypa.io/warnings/backtracking for guidance. If you want to abort this run, press Ctrl + C.\n",
      "  Using cached grpcio_status-1.63.0-py3-none-any.whl.metadata (1.1 kB)\n",
      "  Using cached grpcio_status-1.62.3-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.62.2-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.62.1-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.62.0-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.61.3-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.60.2-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.60.1-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.60.0-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.59.5-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.59.3-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.59.2-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.59.0-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.58.3-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.58.0-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.57.0-py3-none-any.whl.metadata (1.2 kB)\n",
      "  Using cached grpcio_status-1.56.2-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.56.0-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.55.3-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.54.3-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.54.2-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.54.0-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.53.2-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.53.1-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.53.0-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.51.3-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.51.1-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.50.0-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.49.1-py3-none-any.whl.metadata (1.3 kB)\n",
      "  Using cached grpcio_status-1.48.2-py3-none-any.whl.metadata (1.2 kB)\n",
      "Requirement already satisfied: parso<0.9.0,>=0.8.3 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jedi>=0.16->ipython<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.8.3)\n",
      "Collecting deprecated>=1.2.6 (from opentelemetry-api>=1.27.0->google-cloud-pubsub<3,>=2.1.0->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached Deprecated-1.2.14-py2.py3-none-any.whl.metadata (5.4 kB)\n",
      "Collecting importlib-metadata<=8.4.0,>=6.0 (from opentelemetry-api>=1.27.0->google-cloud-pubsub<3,>=2.1.0->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached importlib_metadata-8.4.0-py3-none-any.whl.metadata (4.7 kB)\n",
      "Collecting opentelemetry-semantic-conventions==0.48b0 (from opentelemetry-sdk>=1.27.0->google-cloud-pubsub<3,>=2.1.0->apache-beam[gcp]<3,>=2.47->tfx)\n",
      "  Using cached opentelemetry_semantic_conventions-0.48b0-py3-none-any.whl.metadata (2.4 kB)\n",
      "Requirement already satisfied: wcwidth in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0->ipython<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.2.5)\n",
      "Collecting google-auth-oauthlib<1.1,>=0.5 (from tensorboard<2.14,>=2.13->tensorflow-intel==2.13.1->tensorflow<2.14,>=2.13.0->tfx)\n",
      "  Using cached google_auth_oauthlib-1.0.0-py2.py3-none-any.whl.metadata (2.7 kB)\n",
      "Requirement already satisfied: markdown>=2.6.8 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tensorboard<2.14,>=2.13->tensorflow-intel==2.13.1->tensorflow<2.14,>=2.13.0->tfx) (3.7)\n",
      "Requirement already satisfied: tensorboard-data-server<0.8.0,>=0.7.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tensorboard<2.14,>=2.13->tensorflow-intel==2.13.1->tensorflow<2.14,>=2.13.0->tfx) (0.7.2)\n",
      "Requirement already satisfied: werkzeug>=1.0.1 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from tensorboard<2.14,>=2.13->tensorflow-intel==2.13.1->tensorflow<2.14,>=2.13.0->tfx) (3.0.6)\n",
      "Requirement already satisfied: notebook>=4.4.1 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (7.2.2)\n",
      "Requirement already satisfied: zipp>=0.5 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from importlib-metadata<=8.4.0,>=6.0->opentelemetry-api>=1.27.0->google-cloud-pubsub<3,>=2.1.0->apache-beam[gcp]<3,>=2.47->tfx) (3.20.2)\n",
      "Requirement already satisfied: jupyter-server<3,>=2.4.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (2.14.1)\n",
      "Requirement already satisfied: jupyterlab-server<3,>=2.27.1 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (2.27.3)\n",
      "Requirement already satisfied: jupyterlab<4.3,>=4.2.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (4.2.5)\n",
      "Requirement already satisfied: notebook-shim<0.3,>=0.2 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.2.3)\n",
      "Requirement already satisfied: tornado>=6.2.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (6.4.1)\n",
      "Requirement already satisfied: anyio>=3.1.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (4.6.2)\n",
      "Requirement already satisfied: argon2-cffi>=21.1 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (21.3.0)\n",
      "Requirement already satisfied: jupyter-client>=7.4.4 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (8.6.0)\n",
      "Requirement already satisfied: jupyter-core!=5.0.*,>=4.12 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (5.7.2)\n",
      "Requirement already satisfied: jupyter-events>=0.9.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.10.0)\n",
      "Requirement already satisfied: jupyter-server-terminals>=0.4.4 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.4.4)\n",
      "Requirement already satisfied: nbconvert>=6.4.4 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (7.16.4)\n",
      "Requirement already satisfied: nbformat>=5.3.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (5.10.4)\n",
      "INFO: pip is looking at multiple versions of jupyter-server to determine which version is compatible with other requirements. This could take a while.\n",
      "Collecting jupyter-server<3,>=2.4.0 (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached jupyter_server-2.14.2-py3-none-any.whl.metadata (8.4 kB)\n",
      "  Using cached jupyter_server-2.14.0-py3-none-any.whl.metadata (8.4 kB)\n",
      "  Using cached jupyter_server-2.13.0-py3-none-any.whl.metadata (8.4 kB)\n",
      "Requirement already satisfied: prometheus-client in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.14.1)\n",
      "Requirement already satisfied: pywinpty in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (2.0.10)\n",
      "Requirement already satisfied: pyzmq>=24 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (25.1.2)\n",
      "Requirement already satisfied: send2trash>=1.8.2 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (1.8.2)\n",
      "Requirement already satisfied: terminado>=0.8.3 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.17.1)\n",
      "Requirement already satisfied: async-lru>=1.0.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyterlab<4.3,>=4.2.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (2.0.4)\n",
      "Requirement already satisfied: httpx>=0.25.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyterlab<4.3,>=4.2.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.27.0)\n",
      "Requirement already satisfied: ipykernel>=6.5.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyterlab<4.3,>=4.2.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (6.29.5)\n",
      "Requirement already satisfied: jupyter-lsp>=2.0.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyterlab<4.3,>=4.2.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (2.2.0)\n",
      "Requirement already satisfied: tomli>=1.2.2 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyterlab<4.3,>=4.2.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (2.0.1)\n",
      "Requirement already satisfied: babel>=2.10 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyterlab-server<3,>=2.27.1->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (2.11.0)\n",
      "Requirement already satisfied: json5>=0.9.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyterlab-server<3,>=2.27.1->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.9.6)\n",
      "INFO: pip is looking at multiple versions of jupyterlab-server to determine which version is compatible with other requirements. This could take a while.\n",
      "Collecting jupyterlab-server<3,>=2.27.1 (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached jupyterlab_server-2.27.2-py3-none-any.whl.metadata (5.9 kB)\n",
      "  Using cached jupyterlab_server-2.27.1-py3-none-any.whl.metadata (5.9 kB)\n",
      "Collecting jupyterlab<4.3,>=4.2.0 (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached jupyterlab-4.2.5-py3-none-any.whl.metadata (16 kB)\n",
      "  Using cached jupyterlab-4.2.4-py3-none-any.whl.metadata (16 kB)\n",
      "INFO: pip is still looking at multiple versions of jupyterlab-server to determine which version is compatible with other requirements. This could take a while.\n",
      "  Using cached jupyterlab-4.2.3-py3-none-any.whl.metadata (16 kB)\n",
      "  Using cached jupyterlab-4.2.2-py3-none-any.whl.metadata (16 kB)\n",
      "INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. See https://pip.pypa.io/warnings/backtracking for guidance. If you want to abort this run, press Ctrl + C.\n",
      "  Using cached jupyterlab-4.2.1-py3-none-any.whl.metadata (16 kB)\n",
      "  Using cached jupyterlab-4.2.0-py3-none-any.whl.metadata (16 kB)\n",
      "Collecting jupyter-server<3,>=2.4.0 (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached jupyter_server-2.12.5-py3-none-any.whl.metadata (8.4 kB)\n",
      "INFO: pip is still looking at multiple versions of jupyter-server to determine which version is compatible with other requirements. This could take a while.\n",
      "  Using cached jupyter_server-2.12.4-py3-none-any.whl.metadata (8.4 kB)\n",
      "  Using cached jupyter_server-2.12.3-py3-none-any.whl.metadata (8.4 kB)\n",
      "INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. See https://pip.pypa.io/warnings/backtracking for guidance. If you want to abort this run, press Ctrl + C.\n",
      "  Using cached jupyter_server-2.12.2-py3-none-any.whl.metadata (8.4 kB)\n",
      "  Using cached jupyter_server-2.12.1-py3-none-any.whl.metadata (8.4 kB)\n",
      "  Using cached jupyter_server-2.12.0-py3-none-any.whl.metadata (8.4 kB)\n",
      "  Using cached jupyter_server-2.11.2-py3-none-any.whl.metadata (8.4 kB)\n",
      "  Using cached jupyter_server-2.11.1-py3-none-any.whl.metadata (8.4 kB)\n",
      "  Using cached jupyter_server-2.10.1-py3-none-any.whl.metadata (8.4 kB)\n",
      "  Using cached jupyter_server-2.10.0-py3-none-any.whl.metadata (8.4 kB)\n",
      "  Using cached jupyter_server-2.9.1-py3-none-any.whl.metadata (8.6 kB)\n",
      "  Using cached jupyter_server-2.9.0-py3-none-any.whl.metadata (8.6 kB)\n",
      "  Using cached jupyter_server-2.8.0-py3-none-any.whl.metadata (8.6 kB)\n",
      "  Using cached jupyter_server-2.7.3-py3-none-any.whl.metadata (8.6 kB)\n",
      "  Using cached jupyter_server-2.7.2-py3-none-any.whl.metadata (8.6 kB)\n",
      "  Using cached jupyter_server-2.7.1-py3-none-any.whl.metadata (8.6 kB)\n",
      "  Using cached jupyter_server-2.7.0-py3-none-any.whl.metadata (8.6 kB)\n",
      "  Using cached jupyter_server-2.6.0-py3-none-any.whl.metadata (8.5 kB)\n",
      "  Using cached jupyter_server-2.5.0-py3-none-any.whl.metadata (8.7 kB)\n",
      "  Using cached jupyter_server-2.4.0-py3-none-any.whl.metadata (8.7 kB)\n",
      "Collecting notebook>=4.4.1 (from widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached notebook-7.2.2-py3-none-any.whl.metadata (10 kB)\n",
      "  Using cached notebook-7.2.1-py3-none-any.whl.metadata (10 kB)\n",
      "  Using cached notebook-7.2.0-py3-none-any.whl.metadata (10 kB)\n",
      "  Using cached notebook-7.1.3-py3-none-any.whl.metadata (10 kB)\n",
      "Collecting jupyterlab<4.2,>=4.1.1 (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached jupyterlab-4.1.8-py3-none-any.whl.metadata (16 kB)\n",
      "  Using cached jupyterlab-4.1.7-py3-none-any.whl.metadata (16 kB)\n",
      "  Using cached jupyterlab-4.1.6-py3-none-any.whl.metadata (16 kB)\n",
      "Collecting jupyterlab-server<3,>=2.22.1 (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached jupyterlab_server-2.27.0-py3-none-any.whl.metadata (5.9 kB)\n",
      "  Using cached jupyterlab_server-2.26.0-py3-none-any.whl.metadata (5.9 kB)\n",
      "  Using cached jupyterlab_server-2.25.4-py3-none-any.whl.metadata (5.9 kB)\n",
      "  Using cached jupyterlab_server-2.25.3-py3-none-any.whl.metadata (5.9 kB)\n",
      "  Using cached jupyterlab_server-2.25.2-py3-none-any.whl.metadata (5.9 kB)\n",
      "  Using cached jupyterlab_server-2.25.1-py3-none-any.whl.metadata (5.9 kB)\n",
      "  Using cached jupyterlab_server-2.25.0-py3-none-any.whl.metadata (5.9 kB)\n",
      "  Using cached jupyterlab_server-2.24.0-py3-none-any.whl.metadata (5.8 kB)\n",
      "  Using cached jupyterlab_server-2.23.0-py3-none-any.whl.metadata (5.8 kB)\n",
      "  Using cached jupyterlab_server-2.22.1-py3-none-any.whl.metadata (5.9 kB)\n",
      "Collecting jupyterlab<4.2,>=4.1.1 (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached jupyterlab-4.1.5-py3-none-any.whl.metadata (15 kB)\n",
      "  Using cached jupyterlab-4.1.4-py3-none-any.whl.metadata (15 kB)\n",
      "  Using cached jupyterlab-4.1.3-py3-none-any.whl.metadata (15 kB)\n",
      "  Using cached jupyterlab-4.1.2-py3-none-any.whl.metadata (15 kB)\n",
      "  Using cached jupyterlab-4.1.1-py3-none-any.whl.metadata (15 kB)\n",
      "Collecting notebook>=4.4.1 (from widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached notebook-7.1.2-py3-none-any.whl.metadata (10 kB)\n",
      "  Using cached notebook-7.1.1-py3-none-any.whl.metadata (10 kB)\n",
      "  Using cached notebook-7.1.0-py3-none-any.whl.metadata (10 kB)\n",
      "  Using cached notebook-7.0.8-py3-none-any.whl.metadata (10 kB)\n",
      "Collecting jupyterlab<4.1,>=4.0.2 (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached jupyterlab-4.0.13-py3-none-any.whl.metadata (15 kB)\n",
      "  Using cached jupyterlab-4.0.12-py3-none-any.whl.metadata (15 kB)\n",
      "  Using cached jupyterlab-4.0.11-py3-none-any.whl.metadata (15 kB)\n",
      "  Using cached jupyterlab-4.0.10-py3-none-any.whl.metadata (15 kB)\n",
      "  Using cached jupyterlab-4.0.9-py3-none-any.whl.metadata (15 kB)\n",
      "  Using cached jupyterlab-4.0.8-py3-none-any.whl.metadata (15 kB)\n",
      "  Using cached jupyterlab-4.0.7-py3-none-any.whl.metadata (15 kB)\n",
      "  Using cached jupyterlab-4.0.6-py3-none-any.whl.metadata (15 kB)\n",
      "  Using cached jupyterlab-4.0.5-py3-none-any.whl.metadata (15 kB)\n",
      "  Using cached jupyterlab-4.0.4-py3-none-any.whl.metadata (15 kB)\n",
      "  Using cached jupyterlab-4.0.3-py3-none-any.whl.metadata (15 kB)\n",
      "  Using cached jupyterlab-4.0.2-py3-none-any.whl.metadata (15 kB)\n",
      "Collecting notebook>=4.4.1 (from widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached notebook-7.0.7-py3-none-any.whl.metadata (10 kB)\n",
      "Collecting jupyterlab<5,>=4.0.2 (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached jupyterlab-4.3.0-py3-none-any.whl.metadata (16 kB)\n",
      "  Using cached jupyterlab-4.1.0-py3-none-any.whl.metadata (15 kB)\n",
      "Collecting notebook>=4.4.1 (from widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached notebook-7.0.6-py3-none-any.whl.metadata (10 kB)\n",
      "  Using cached notebook-7.0.5-py3-none-any.whl.metadata (10 kB)\n",
      "  Using cached notebook-7.0.4-py3-none-any.whl.metadata (10 kB)\n",
      "  Using cached notebook-7.0.3-py3-none-any.whl.metadata (10 kB)\n",
      "  Using cached notebook-7.0.2-py3-none-any.whl.metadata (10 kB)\n",
      "  Using cached notebook-7.0.1-py3-none-any.whl.metadata (10 kB)\n",
      "  Using cached notebook-7.0.0-py3-none-any.whl.metadata (10 kB)\n",
      "  Using cached notebook-6.5.7-py3-none-any.whl.metadata (2.5 kB)\n",
      "Collecting jupyter-client<8,>=5.3.4 (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached jupyter_client-7.4.9-py3-none-any.whl.metadata (8.5 kB)\n",
      "Requirement already satisfied: nest-asyncio>=1.5 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (1.6.0)\n",
      "Collecting nbclassic>=0.4.7 (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached nbclassic-1.1.0-py3-none-any.whl.metadata (3.6 kB)\n",
      "Collecting entrypoints (from jupyter-client<8,>=5.3.4->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached entrypoints-0.4-py3-none-any.whl.metadata (2.6 kB)\n",
      "Requirement already satisfied: platformdirs>=2.5 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-core!=5.0.*,>=4.12->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (3.10.0)\n",
      "Requirement already satisfied: beautifulsoup4 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from nbconvert>=6.4.4->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (4.12.3)\n",
      "Requirement already satisfied: bleach!=5.0.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from nbconvert>=6.4.4->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (4.1.0)\n",
      "Requirement already satisfied: defusedxml in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from nbconvert>=6.4.4->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.7.1)\n",
      "Requirement already satisfied: jupyterlab-pygments in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from nbconvert>=6.4.4->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.1.2)\n",
      "Requirement already satisfied: mistune<4,>=2.0.3 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from nbconvert>=6.4.4->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (2.0.4)\n",
      "Requirement already satisfied: nbclient>=0.5.0 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from nbconvert>=6.4.4->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.8.0)\n",
      "Requirement already satisfied: pandocfilters>=1.4.1 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from nbconvert>=6.4.4->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (1.5.0)\n",
      "Requirement already satisfied: tinycss2 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from nbconvert>=6.4.4->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (1.2.1)\n",
      "Requirement already satisfied: fastjsonschema>=2.15 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from nbformat>=5.3.0->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (2.16.2)\n",
      "Requirement already satisfied: argon2-cffi-bindings in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from argon2-cffi>=21.1->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (21.2.0)\n",
      "Requirement already satisfied: debugpy>=1.6.5 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from ipykernel>=6.5.0->jupyterlab<4.3,>=4.2.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (1.6.7)\n",
      "Requirement already satisfied: webencodings in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from bleach!=5.0.0->nbconvert>=6.4.4->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.5.1)\n",
      "INFO: pip is looking at multiple versions of jsonschema to determine which version is compatible with other requirements. This could take a while.\n",
      "Collecting jsonschema>=2.6 (from nbformat>=5.3.0->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached jsonschema-4.22.0-py3-none-any.whl.metadata (8.2 kB)\n",
      "  Using cached jsonschema-4.21.1-py3-none-any.whl.metadata (7.8 kB)\n",
      "  Using cached jsonschema-4.21.0-py3-none-any.whl.metadata (8.0 kB)\n",
      "  Using cached jsonschema-4.20.0-py3-none-any.whl.metadata (8.1 kB)\n",
      "  Using cached jsonschema-4.19.2-py3-none-any.whl.metadata (7.9 kB)\n",
      "  Using cached jsonschema-4.19.1-py3-none-any.whl.metadata (7.9 kB)\n",
      "  Using cached jsonschema-4.19.0-py3-none-any.whl.metadata (8.2 kB)\n",
      "INFO: pip is still looking at multiple versions of jsonschema to determine which version is compatible with other requirements. This could take a while.\n",
      "  Using cached jsonschema-4.18.6-py3-none-any.whl.metadata (7.8 kB)\n",
      "  Using cached jsonschema-4.18.5-py3-none-any.whl.metadata (7.7 kB)\n",
      "  Using cached jsonschema-4.18.4-py3-none-any.whl.metadata (7.8 kB)\n",
      "  Using cached jsonschema-4.18.3-py3-none-any.whl.metadata (7.9 kB)\n",
      "  Using cached jsonschema-4.18.2-py3-none-any.whl.metadata (7.8 kB)\n",
      "INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. See https://pip.pypa.io/warnings/backtracking for guidance. If you want to abort this run, press Ctrl + C.\n",
      "  Using cached jsonschema-4.18.1-py3-none-any.whl.metadata (7.8 kB)\n",
      "  Using cached jsonschema-4.18.0-py3-none-any.whl.metadata (10 kB)\n",
      "  Using cached jsonschema-4.17.3-py3-none-any.whl.metadata (7.9 kB)\n",
      "Collecting pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 (from jsonschema>=2.6->nbformat>=5.3.0->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached pyrsistent-0.20.0-cp310-cp310-win_amd64.whl.metadata (976 bytes)\n",
      "Requirement already satisfied: cffi>=1.0.1 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from argon2-cffi-bindings->argon2-cffi>=21.1->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (1.17.1)\n",
      "Requirement already satisfied: soupsieve>1.2 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from beautifulsoup4->nbconvert>=6.4.4->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (2.5)\n",
      "Requirement already satisfied: pycparser in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from cffi>=1.0.1->argon2-cffi-bindings->argon2-cffi>=21.1->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (2.21)\n",
      "Requirement already satisfied: sniffio>=1.1 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from anyio>=3.1.0->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (1.3.0)\n",
      "Requirement already satisfied: exceptiongroup>=1.0.2 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from anyio>=3.1.0->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (1.2.0)\n",
      "Requirement already satisfied: python-json-logger>=2.0.4 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-events>=0.9.0->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (2.0.7)\n",
      "Requirement already satisfied: referencing in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-events>=0.9.0->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.30.2)\n",
      "Requirement already satisfied: rfc3339-validator in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-events>=0.9.0->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.1.4)\n",
      "Requirement already satisfied: rfc3986-validator>=0.1.1 in c:\\users\\rohan\\anaconda3\\envs\\medscript-ai\\lib\\site-packages (from jupyter-events>=0.9.0->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx) (0.1.1)\n",
      "Collecting jupyter-events>=0.9.0 (from jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached jupyter_events-0.10.0-py3-none-any.whl.metadata (5.9 kB)\n",
      "  Using cached jupyter_events-0.9.1-py3-none-any.whl.metadata (5.7 kB)\n",
      "  Using cached jupyter_events-0.9.0-py3-none-any.whl.metadata (5.7 kB)\n",
      "Collecting jsonschema>=2.6 (from nbformat>=5.3.0->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached jsonschema-4.17.1-py3-none-any.whl.metadata (7.9 kB)\n",
      "  Using cached jsonschema-4.17.0-py3-none-any.whl.metadata (8.4 kB)\n",
      "  Using cached jsonschema-4.16.0-py3-none-any.whl.metadata (7.8 kB)\n",
      "  Using cached jsonschema-4.15.0-py3-none-any.whl.metadata (7.8 kB)\n",
      "  Using cached jsonschema-4.14.0-py3-none-any.whl.metadata (7.6 kB)\n",
      "  Using cached jsonschema-4.13.0-py3-none-any.whl.metadata (7.4 kB)\n",
      "  Using cached jsonschema-4.12.1-py3-none-any.whl.metadata (7.0 kB)\n",
      "  Using cached jsonschema-4.12.0-py3-none-any.whl.metadata (7.0 kB)\n",
      "  Using cached jsonschema-4.11.0-py3-none-any.whl.metadata (7.0 kB)\n",
      "  Using cached jsonschema-4.10.3-py3-none-any.whl.metadata (8.0 kB)\n",
      "  Using cached jsonschema-4.10.2-py3-none-any.whl.metadata (8.0 kB)\n",
      "  Using cached jsonschema-4.10.1-py3-none-any.whl.metadata (8.0 kB)\n",
      "  Using cached jsonschema-4.10.0-py3-none-any.whl.metadata (8.0 kB)\n",
      "  Using cached jsonschema-4.9.1-py3-none-any.whl.metadata (8.0 kB)\n",
      "  Using cached jsonschema-4.9.0-py3-none-any.whl.metadata (8.0 kB)\n",
      "  Using cached jsonschema-4.8.0-py3-none-any.whl.metadata (7.9 kB)\n",
      "  Using cached jsonschema-4.7.2-py3-none-any.whl.metadata (7.9 kB)\n",
      "  Using cached jsonschema-4.7.1-py3-none-any.whl.metadata (7.9 kB)\n",
      "  Using cached jsonschema-4.7.0-py3-none-any.whl.metadata (7.9 kB)\n",
      "  Using cached jsonschema-4.6.2-py3-none-any.whl.metadata (7.9 kB)\n",
      "  Using cached jsonschema-4.6.1-py3-none-any.whl.metadata (7.9 kB)\n",
      "  Using cached jsonschema-4.6.0-py3-none-any.whl.metadata (7.9 kB)\n",
      "  Using cached jsonschema-4.5.1-py3-none-any.whl.metadata (7.8 kB)\n",
      "  Using cached jsonschema-4.5.0-py3-none-any.whl.metadata (7.8 kB)\n",
      "  Using cached jsonschema-4.4.0-py3-none-any.whl.metadata (7.5 kB)\n",
      "  Using cached jsonschema-4.3.3-py3-none-any.whl.metadata (7.5 kB)\n",
      "  Using cached jsonschema-4.3.2-py3-none-any.whl.metadata (7.5 kB)\n",
      "  Using cached jsonschema-4.3.1-py3-none-any.whl.metadata (7.5 kB)\n",
      "  Using cached jsonschema-4.3.0-py3-none-any.whl.metadata (7.5 kB)\n",
      "  Using cached jsonschema-4.2.1-py3-none-any.whl.metadata (7.4 kB)\n",
      "  Using cached jsonschema-4.2.0-py3-none-any.whl.metadata (7.4 kB)\n",
      "  Using cached jsonschema-4.1.2-py3-none-any.whl.metadata (7.2 kB)\n",
      "  Using cached jsonschema-4.1.1-py3-none-any.whl.metadata (7.2 kB)\n",
      "  Using cached jsonschema-4.1.0-py3-none-any.whl.metadata (7.2 kB)\n",
      "  Using cached jsonschema-4.0.1-py3-none-any.whl.metadata (7.1 kB)\n",
      "  Using cached jsonschema-3.2.0-py2.py3-none-any.whl.metadata (7.8 kB)\n",
      "  Using cached jsonschema-3.1.1-py2.py3-none-any.whl.metadata (7.4 kB)\n",
      "  Using cached jsonschema-3.1.0-py2.py3-none-any.whl.metadata (7.4 kB)\n",
      "Collecting js-regex>=1.0.0 (from jsonschema>=2.6->nbformat>=5.3.0->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached js_regex-1.0.1-py3-none-any.whl.metadata (6.2 kB)\n",
      "Collecting jsonschema>=2.6 (from nbformat>=5.3.0->jupyter-server<3,>=2.4.0->notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached jsonschema-3.0.2-py2.py3-none-any.whl.metadata (7.4 kB)\n",
      "  Using cached jsonschema-3.0.1-py2.py3-none-any.whl.metadata (6.2 kB)\n",
      "  Using cached jsonschema-3.0.0-py2.py3-none-any.whl.metadata (6.2 kB)\n",
      "  Using cached jsonschema-2.6.0-py2.py3-none-any.whl.metadata (4.6 kB)\n",
      "Collecting nbformat (from notebook>=4.4.1->widgetsnbextension~=3.6.10->ipywidgets<8,>=7->tensorflow-model-analysis<0.46.0,>=0.45.0->tfx)\n",
      "  Using cached nbformat-5.10.4-py3-none-any.whl.metadata (3.6 kB)\n",
      "  Using cached nbformat-5.10.3-py3-none-any.whl.metadata (3.6 kB)\n",
      "  Using cached nbformat-5.10.2-py3-none-any.whl.metadata (3.6 kB)\n",
      "  Using cached nbformat-5.10.1-py3-none-any.whl.metadata (3.6 kB)\n",
      "  Using cached nbformat-5.10.0-py3-none-any.whl.metadata (3.6 kB)\n",
      "  Using cached nbformat-5.9.2-py3-none-any.whl.metadata (3.4 kB)\n",
      "  Using cached nbformat-5.9.1-py3-none-any.whl.metadata (3.3 kB)\n",
      "  Using cached nbformat-5.9.0-py3-none-any.whl.metadata (3.3 kB)\n",
      "  Using cached nbformat-5.8.0-py3-none-any.whl.metadata (3.5 kB)\n",
      "  Using cached nbformat-5.7.3-py3-none-any.whl.metadata (5.0 kB)\n",
      "  Using cached nbformat-5.7.2-py3-none-any.whl.metadata (5.0 kB)\n",
      "  Using cached nbformat-5.7.1-py3-none-any.whl.metadata (5.0 kB)\n",
      "  Using cached nbformat-5.7.0-py3-none-any.whl.metadata (4.7 kB)\n",
      "Collecting tfx\n",
      "  Using cached tfx-0.23.1-py3-none-any.whl.metadata (9.4 kB)\n",
      "Collecting absl-py<0.9,>=0.7 (from tfx)\n",
      "  Using cached absl-py-0.8.1.tar.gz (103 kB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  error: subprocess-exited-with-error\n",
      "  \n",
      "  python setup.py egg_info did not run successfully.\n",
      "  exit code: 1\n",
      "  \n",
      "  [6 lines of output]\n",
      "  Traceback (most recent call last):\n",
      "    File \"<string>\", line 2, in <module>\n",
      "    File \"<pip-setuptools-caller>\", line 34, in <module>\n",
      "    File \"C:\\Users\\rohan\\AppData\\Local\\Temp\\pip-install-9zuks6ix\\absl-py_ca753f1eaca6440cb33cdf527d2e243d\\setup.py\", line 34, in <module>\n",
      "      raise RuntimeError('Python version 2.7 or 3.4+ is required.')\n",
      "  RuntimeError: Python version 2.7 or 3.4+ is required.\n",
      "  [end of output]\n",
      "  \n",
      "  note: This error originates from a subprocess, and is likely not a problem with pip.\n",
      "error: metadata-generation-failed\n",
      "\n",
      "Encountered error while generating package metadata.\n",
      "\n",
      "See above for output.\n",
      "\n",
      "note: This is an issue with the package mentioned above, not pip.\n",
      "hint: See above for details.\n"
     ]
    }
   ],
   "execution_count": 7
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
